lib/csf-tools/src/getStorySortParameter.test.ts

```diff
@@ -0,0 +1,156 @@
+describe('getStorySortParameter', () => {
+  it('no parameters', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const decorators = [];
+      `)
+    ).toBeUndefined();
+  });
+
+  it('invalid parameters', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = [];
+      `)
+    ).toBeUndefined();
+  });
+
+  it('no storySort parameter', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = {
+          layout: 'fullscreen',
+        };
+      `)
+    ).toBeUndefined();
+  });
+
+  it('with wildcards', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = {
+          options: {
+            storySort: [
+              "Intro",
+              "Pages",
+              ["Home", "Login", "Admin"],
+              "Components",
+              "*",
+              "WIP",    
+            ]
+          }
+        };
+      `)
+    ).toMatchInlineSnapshot(`
+      Array [
+        "Intro",
+        "Pages",
+        Array [
+          "Home",
+          "Login",
+          "Admin",
+        ],
+        "Components",
+        "*",
+        "WIP",
+      ]
+    `);
+  });
+
+  it('arrow function', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = {
+          options: {
+            storySort: (a, b) =>
+              a[1].kind === b[1].kind
+                ? 0
+                : a[1].id.localeCompare(b[1].id, undefined, { numeric: true }),
+          },
+        };
+      `)
+    ).toMatchInlineSnapshot(`[Function]`);
+  });
+
+  it('function', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = {
+          options: {
+            storySort: function sortStories(a, b) {
+              return a[1].kind === b[1].kind
+                ? 0
+                : a[1].id.localeCompare(b[1].id, undefined, { numeric: true });
+            },
+          },
+        };
+      `)
+    ).toMatchInlineSnapshot(`[Function]`);
+  });
+
+  it('empty sort', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = {
+          options: {
+            storySort: {
+              method: "",
+              order: [],
+              locales: "",
+            },
+          },
+        };
+      `)
+    ).toMatchInlineSnapshot(`
+      Object {
+        "locales": "",
+        "method": "",
+        "order": Array [],
+      }
+    `);
+  });
+
+  it('parameters typescript', () => {
+    expect(
+      getStorySortParameter(dedent`
+        export const parameters = {
+          options: {
+            storySort: {
+              method: "",
+              order: [],
+              locales: "",
+            },
+          },
+        } as Parameters;
+      `)
+    ).toMatchInlineSnapshot(`
+      Object {
+        "locales": "",
+        "method": "",
+        "order": Array [],
+      }
+    `);
+  });
+
+  // unsupported
```

Comment:
Is this a TODO, or are we not supporting this?

Reply:
Also, what about various other annoying combinations, like:

```js
function storySort(..) {};

export const parameters = { options: { storySort } };
```

or

```js
export const parameters = {
  options: {
    storySort() {
    }
  }
}
```

or, even the slightly evil:

```js
const options = { storySort: () => { ... } };

export const parameters = { options };
```

Reply:
I included this as a placeholder for discussion. Where do we draw the line?

Reply:
Updated the parsing to draw the line explicitly and added tests @tmeasday